src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.cs

```diff
@@ -204,6 +229,52 @@ protected override void Free()
             base.Free();
         }
 
+        protected override bool TryGetVariable(VariableIdentifier identifier, out int slot)
+        {
+            return _variableSlot.TryGetValue(identifier, out slot);
+        }
+
+        protected override int AddVariable(VariableIdentifier identifier)
+        {
+            int slot = nextVariableSlot++;
+            _variableSlot.Add(identifier, slot);
+            if (slot >= variableBySlot.Length)
+            {
+                Array.Resize(ref this.variableBySlot, slot * 2);
```

Comment:
I know this stuff is just copied, but do we want to consider just making `variableBySlot` an `ArrayBuilder`, which handles this already?